conans/util/files.py

```diff
@@ -189,6 +189,10 @@ def save(path, content, only_if_modified=False, encoding="utf-8"):
     """
     try:
         os.makedirs(os.path.dirname(path))
+    except OSError as error:
+        if error.errno == errno.EACCES:
```

Comment:
I think this has some potential to break something. The folder might exist, but created by another user, and instead of getting a EEXIST error, get an EACCES error? Who knows.

I think I would make sense to check the existence of the folder, something like:

```python
folder = os.path.dirname(path)
if not os.path.isdir(folder):
     try:
         os.makedirs(folder)
     except Exception as e:
         raise OSError("The folder {} does not exist and could not be created: {}".format(folder, str(e)))
```

Reply:
On Linux I'm sure it will result in a Permission error (errno.EACCES) if I try to re-create a folder which already exists, but created by another user. Anyway, I think don't harm using your logic, because I can say nothing about OSX and Windows behavior for this case.

Reply:
Done.